examples/basic_cog.py

```diff
@@ -0,0 +1,116 @@
+        self.afk = {}
+
+    # Cogs also have the capability to define checks that apply to every command in the Cog.
+    # They work similarly to regular checks elsewhere in the framework.
+    # ref: https://discordpy.readthedocs.io/en/latest/ext/commands/commands.html#checks
+    async def cog_check(self, ctx):
+        # Check if the command was invoked within a Guild channel.
+        return ctx.guild is not None
+
+    # Commands within a Cog are defined using the commands.command decorator
+    # instead of the bot.command decorator.
+    # They must also pass a `self` parameter since it's within a class.
+    @commands.command()
+    async def hello(self, ctx):
+        """Adds two numbers together"""
+        await ctx.send("Hello am I an example bot!")
+
+    # A command group within a Cog.
+    # The `invoke_without_command` flag for the group allows us to use the parent of the group as a command.
+    @commands.group(invoke_without_command=True)
+    async def afk(self, ctx, *, reason: str):
+        """
+        Set your afk status
+        """
+        u_id = ctx.author.id
+        # Simply check if they exist in our afk dictionary aldready.
+        # If they exist then tell them they are afk, or let them set a new afk status.
+        rating = self.afk.get(u_id)
+        if not rating:
+            self.afk[u_id] = reason
+            await ctx.send(f"{ctx.author.mention} I have set your afk status as `{reason}`.")
+        else:
+            await ctx.send(f"{ctx.author.mdention} you are aldready afk.")
+
+    # Subcommand for the group
+    @afk.command(name='user')
+    async def _user(self, ctx, *, user: discord.User = None):
+        """Check a user's afk status"""
+        if not user:
+            return await ctx.send("I need a user to check the afk status")
+        afk = self.afk.get(user.id)
+        if afk:
+            return await ctx.send(f"{user.name} is afk with reason `{afk}`")
+        return await ctx.send(f'{user.name} is currently not afk')
+
+    # Listening to events within a cog requires the `commands.Cog.listener` decorator.
+    # Note that `self` has to be passed for these as well.
+    @commands.Cog.listener()
+    async def on_message(self, message):
+        # Don't want to reply to bots
+        if message.author.bot:
+            return
+        # Ensure that the message wasn't the afk command
+        ctx = await self.bot.get_context(message)
+        if ctx.invoked_with == "afk":
+            return
+
+        # If an afk user has sent a message, remove their afk
+        m_id = message.author.id
+        afk_status = self.afk.get(m_id)
+        if afk_status:
+            await message.channel.send(f"Welcome back {message.author.mention}! I have removed your afk status.")
+            self.afk.pop(m_id)
+            return
+
+        # If an afk user is mentioned in a message, let the author of the message know they are afk
+        mentions = [user.id for user in message.mentions]
```

Comment:
```suggestion
        mentions = {user.id: user for user in message.mentions}
```

Dicts have faster lookup than lists. Also this will be used later on for resolving.